src/simpleNbox.cpp

```diff
@@ -623,12 +750,14 @@ void SimpleNbox::stashCValues( double t, const double c[] )
     H_LOG( logger,Logger::DEBUG ) << "veg_delta = " << veg_delta << std::endl;
     H_LOG( logger,Logger::DEBUG ) << "det_delta = " << det_delta << std::endl;
     H_LOG( logger,Logger::DEBUG ) << "soil_delta = " << soil_delta << std::endl;
-    for( unitval_stringmap::const_iterator it = veg_c.begin(); it != veg_c.end(); it++ ) {
-        std::string biome = it->first;
+
+    std::vector<std::string>::const_iterator it;
+    for( it = biome_list.begin(); it != biome_list.end(); it++ ) {
+        std::string biome = *it;
         const double wt     = ( npp( biome ) + rh( biome ) ) / npp_rh_total;
-        veg_c[ biome ]      = veg_c[ biome ] + veg_delta * wt;
-        detritus_c[ biome ] = detritus_c[ biome ] + det_delta * wt;
-        soil_c[ biome ]     = soil_c[ biome ] + soil_delta * wt;
+        veg_c[ biome ]      = veg_c.at( biome ) + veg_delta * wt;
```

Comment:
I switched to `at` to ensure that this statement threw an error if `biome` didn't exist, now and at all points in the future (i.e. if the earlier checks in this block were removed). But I'm happy to change it back if there's a performance hit of `.at` vs. `[]`.

Changing this to `+=` fails because of the error `no viable overloaded +=`. I took a stab at implementing the overloaded `+=` operator in `unitval.hpp`, but it's sufficiently different from the binary operators that I wasn't totally sure how to proceed.

Reply:
`at` and `[]` do the same thing; it's just that `[]` creates the entry if it doesn't exist already.  `x[y]` seems more natural to me than `x.at(y)`, if you don't need the existence check, but it's fine either way.  

It would be good to have a `+=` operator for unitval objects.  I might have time to crank that out sometime this afternoon.  If not, then we can just leave it as it is.